src/Adapter/Swoole/Internal/SwoolePromise.php

```diff
@@ -0,0 +1,246 @@
+     */
+    public function then(?callable $onFulfilled = null, ?callable $onRejected = null): SwoolePromise
+    {
+        return self::create(function (callable $resolve, callable $reject) use ($onFulfilled, $onRejected) {
+            $this->resolvedCallback = function() use ($resolve, $reject, $onFulfilled, $onRejected) {
+                $callable = $this->isFulfilled() ? $onFulfilled : $onRejected;
+                if (!is_callable($callable)) {
+                    $resolve($this->result);
+                    return;
+                }
+                try {
+                    $resolve($callable($this->result));
+                } catch (\Throwable $error) {
+                    $reject($error);
+                }
+            };
+            if(!$this->isPending()) {
+                ($this->resolvedCallback)();
+            }
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param iterable|SwoolePromise[] $promises
+     * @return SwoolePromise
+     */
+    public static function all(iterable $promises): SwoolePromise
+    {
+        return self::create(function (callable $resolve, callable $reject) use ($promises) {
+            $ticks = count($promises);
+
+            $firstError = null;
+            $channel    = new Channel($ticks);
+            $result     = [];
+            $key        = 0;
+            foreach ($promises as $promise) {
+                if (!$promise instanceof SwoolePromise) {
+                    $channel->close();
+                    throw new RuntimeException(
+                        'Supported only SwoolePromise instance'
+                    );
+                }
+                $promise->then(function ($value) use ($key, &$result, $channel) {
+                    $result[$key] = $value;
+                    $channel->push(true);
+                    return $value;
+                }, function ($error) use ($channel, &$firstError) {
+                    $channel->push(true);
+                    if ($firstError === null) {
+                        $firstError = $error;
+                    }
+                });
+                $key++;
+            }
+            while ($ticks--) {
+                $channel->pop();
+            }
+            $channel->close();
+
+            if ($firstError !== null) {
+                $reject($firstError);
+                return;
+            }
+
+            $resolve($result);
+        });
+    }
+
+    /**
+     * Set resolved result
+     *
+     * @param mixed $value
+     * @return void
+     */
+    private function setResult($value, $state): void
+    {
+        if ($value instanceof self) {
+            $resolved = false;
+            $callable = function ($value) use (&$resolved) {
+                $this->setResult($value);
+                $resolved = true;
+            };
+            $value->then($callable, $callable);
+            // resolve async locking error (code to remove)
+            //while (!$resolved) {
```

Comment:
Now I got non blocking tests `bin/phpunit tests/Adapter/Swoole/EventLoopTest.php`.
I think this part should be somehow adapted to transform `usleep` to callback way.